tests/test_inspector.py

```diff
@@ -0,0 +1,191 @@
+from unittest import TestCase
+from shutil import rmtree
+from tempfile import mkdtemp
+from uuid import uuid4
+from datetime import datetime
+from pathlib import Path
+
+import pynwb
+
+from nwbinspector.refactor_inspector import inspect_nwb
+
+
+class TestInspector(TestCase):
+    def setUp(self):
+        self.tempdir = Path(mkdtemp())
+        self.nwbfile = pynwb.NWBFile(
+            session_description="Testing inspector.",
+            identifier=str(uuid4()),
+            session_start_time=datetime.now().astimezone(),
+        )
+
+    def tearDown(self):
+        rmtree(self.tempdir)
+
+    def add_big_dataset_no_compression(self):
+        device = self.nwbfile.create_device(name="test_device")
+        electrode_group = self.nwbfile.create_electrode_group(
+            name="test_group", description="", location="", device=device
+        )
+        self.num_electrodes = 4
+        electrode_ids = list(range(self.num_electrodes))
+        for id in electrode_ids:
+            self.nwbfile.add_electrode(
+                id=id,
+                x=np.nan,
+                y=np.nan,
+                z=np.nan,
+                imp=np.nan,
+                location="",
+                filtering="",
+                group=electrode_group,
+            )
+        self.electrode_table_region = self.nwbfile.create_electrode_table_region(
+            electrode_ids, description=""
+        )
+
+        n_bytes = 3e6
+        # itemsize of 8 because of float dtype
+        n_frames = int(n_bytes / (self.num_electrodes * 8))
+        ephys_data = np.zeros(shape=(n_frames, self.num_electrodes))
+        ephys_ts = pynwb.ecephys.ElectricalSeries(
+            name="test_ecephys_1",
+            data=ephys_data,
+            electrodes=self.electrode_table_region,
+            rate=10.0,
+        )
+        self.nwbfile.add_acquisition(ephys_ts)
+
+    def add_regular_timestamps(self):
+        time_series = pynwb.ecephys.ElectricalSeries(
+            name="test_ecephys_2",
+            data=np.zeros(shape=(5, self.num_electrodes)),
+            electrodes=self.electrode_table_region,
+            timestamps=np.arange(1.2, 11.2, 2),
+        )
+        self.nwbfile.add_acquisition(time_series)
+
+    def add_flipped_data_orientation(self):
+        time_series = pynwb.ecephys.ElectricalSeries(
+            name="test_ecephys_3",
+            data=np.zeros(shape=(self.num_electrodes, 5)),
+            electrodes=self.electrode_table_region,
+            rate=1.0,
+        )
+        self.nwbfile.add_acquisition(time_series)
+
+    def add_non_matching_timestamps_dimension(self):
+        time_series = pynwb.ecephys.ElectricalSeries(
+            name="test_ecephys_4",
+            data=np.zeros(shape=(self.num_electrodes, 5)),
+            electrodes=self.electrode_table_region,
+            timestamps=np.arange(1.1, 6.1),
+        )
+        self.nwbfile.add_acquisition(time_series)
+
+    def test_inspect_nwb(self):
```

Comment:
Let's try to keep the test to small modular unit tests for now

Reply:
What kind of modularity did you have in mind for that?

Reply:
I mean don't have one big test that tests a bunch of checks all at once. When testing the checks themselves, a single test should test a single check. This would be a prober "unit test" as opposed to an "integration test" which would run more of a simulation of a real file like this is

Reply:
Gotcha